mods/modern-notepad-to-classic-editor.wh.cpp

```diff
@@ -0,0 +1,429 @@
+  }
+
+  // Non-elevated launch: build full command line with classic exe + original args
+  std::wstring newCmd = L"\"" + classicPath + L"\"";
+  {
+    std::wstring argTail = BuildArgsOnly();
+    if (!argTail.empty()) {
+      newCmd += L" ";
+      newCmd += argTail;
+    }
+  }
+
+  std::unique_ptr<wchar_t[]> buf(new wchar_t[newCmd.size() + 1]);
+  wcscpy_s(buf.get(), newCmd.size() + 1, newCmd.c_str());
+
+  STARTUPINFOW si = { 0 }; si.cb = sizeof(si);
+  PROCESS_INFORMATION pi = { 0 };
+
+  if (!CreateProcessW(NULL, buf.get(), NULL, NULL, FALSE, 0, NULL, NULL, &si, &pi)) {
+    InterlockedExchange(&g_handoffFinished, 1);
+    ExitProcess(0);
+  }
+
+  if (g_waitForClassic) {
+    WaitForSingleObject(pi.hProcess, INFINITE);
+    DWORD exitCode = 0;
+    GetExitCodeProcess(pi.hProcess, &exitCode);
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+    InterlockedExchange(&g_handoffFinished, 1);
+    ExitProcess(exitCode);
+  } else {
+    CloseHandle(pi.hProcess);
+    CloseHandle(pi.hThread);
+    InterlockedExchange(&g_handoffFinished, 1);
+    ExitProcess(0);
+  }
+
+  return 0;
+}
+
+// ---------------------------------------------------------------------
+// Windhawk entry points
+// ---------------------------------------------------------------------
+BOOL Wh_ModInit() {
+  // Load settings
+  {
+    std::wstring pathSetting = Wh_GetStringSetting(L"classicPath");
+    if (pathSetting.empty()) pathSetting = L"%SystemRoot%\\System32\\notepad.exe";
+    g_classicPathSetting = pathSetting;
+
+    g_ctrlShiftElevate = Wh_GetIntSetting(L"ctrlShiftElevate", 1) != 0;
+    g_waitForClassic   = Wh_GetIntSetting(L"waitForClassic", 1) != 0;
+  }
+
+  // Only act inside the WindowsApps Notepad process
+  std::wstring cur = GetCurrentExePath();
+  if (!IsModernWindowsAppsNotepadPath(cur)) {
+    return FALSE;
+  }
+
+  // Detect whether our own first real arg starts with /SESSION, exit quietly with 0
+  auto argv = ParseCommandLineW(GetCommandLineW());
+  bool isSessionLaunch = ArgsContainSession(argv);
+
+  // Check for app execution alias
+  {
+    HKEY hKey;
+    if (RegOpenKeyExW(HKEY_CURRENT_USER, L"Software\\Microsoft\\Windows\\CurrentVersion\\App Paths\\notepad.exe", 0, KEY_READ | KEY_WRITE, &hKey) == ERROR_SUCCESS) {
+      WCHAR value[1024] = {};
+      DWORD size = sizeof(value);
+      DWORD type;
+      if (RegQueryValueExW(hKey, NULL, NULL, &type, (BYTE*)value, &size) == ERROR_SUCCESS && type == REG_SZ) {
+        std::wstring val(value);
+        if (val.find(L"WindowsApps") != std::wstring::npos) {
+          int result = isSessionLaunch ? IDNO : MessageBoxW(NULL, L"It seems like an app execution alias for notepad.exe to the modern Notepad is set in Windows settings and this mod cannot work while the alias is enabled. Do you want to disable this alias?", L"Redirect Modern Notepad to Classic Editor (Windhawk Mod)", MB_YESNO | MB_ICONQUESTION);
+          if (result == IDYES) {
+            RegCloseKey(hKey);
```

Comment:
Here, `RegCloseKey(hKey)` will be called twice.

Reply:
Fixed.